frontend/common/services/useFeatureState.ts

```diff
@@ -0,0 +1,79 @@
+import { FeatureState, PagedResponse, Res } from 'common/types/responses'
+import { Req } from 'common/types/requests'
+import { service } from 'common/service'
+import Utils from 'common/utils/utils'
+import { getFeatureSegment } from './useFeatureSegment'
+import { getStore } from 'common/store'
+const _data = require('../data/base/_data')
+export const addFeatureSegmentsToFeatureStates = async (v) => {
+  if (!v.feature_segment) {
+    return v
+  }
+  const featureSegmentData = await getFeatureSegment(getStore(), {
+    id: v.feature_segment,
+  })
+  return {
+    ...v,
+    feature_segment: featureSegmentData.data,
+  }
+}
+export const featureStateService = service
+  .enhanceEndpoints({ addTagTypes: ['FeatureState'] })
+  .injectEndpoints({
+    endpoints: (builder) => ({
+      getFeatureStates: builder.query<
+        Res['featureStates'],
+        Req['getFeatureStates']
+      >({
+        providesTags: [{ id: 'LIST', type: 'FeatureState' }],
+        queryFn: async (query, baseQueryApi, extraOptions, baseQuery) => {
+          //This endpoint returns feature_segments as a number, so it fetches the feature segments and appends
+          const {
+            data,
+          }: {
+            data: PagedResponse<
+              Omit<FeatureState, 'feature_segment'> & {
+                feature_segment: number | null
+              }
+            >
+          } = await baseQuery({
+            url: `features/featurestates/?${Utils.toParam(query)}`,
```

Comment:
I don't _think_ GET requests will be an issue here, but it's probably safest to add the trailing slash here for when this endpoint is used on the combined image. 

```suggestion
            url: `features/featurestates/?${Utils.toParam(query)}/`,
```